### ISSUE TITLE
Replace headers by a given line

### ISSUE THREAD

**Issue by yohanboniface** [gristlabs/grist-core#398]
Hi,

When a Excel file as noisy first rows, it happens that Grist fails on guessing the headers line. Time ago, we've looked at the back code that handle this part, and arrived at the conclusion that we cannot fix this once and for all (trying to have less false negative will increase false positive, and so on).

But instead, one thing that could help: right-click on a line and "Make headers from this line", that would replace the current headers by the values of that given line.

That would serve as a catch all for any weird case we'll never be able to deal with.

Thoughts ? :)

Yohan

**Comment by dsagal** [comment]
@yohanboniface, do you mean an option in any table view, or only in the import dialog?

Another possibility to help this is if we add an explicit option in the import dialog. The code makes its guess of headers, which determines the headers in the preview. We could also show this guess explicitly as an option, like a `Headers from row` with a textbox that contains the number of the guessed row with headers (or `None` if we guessed no headers). The user could then change this option, and see the updated preview, before completing the import.

**Comment by yohanboniface** [comment]
Yes, an option in the import dialog would indeed help. The thing is that when you create a new document by importing a file, you do not have this import dialog, afaik. You only have it when importing a file in a already existing document. That's way I was thinking about an action right from the final table.
